js/stdlib/ts/ALGO.ts

```diff
@@ -2248,6 +2281,8 @@ export const getNetworkTime = async (): Promise<BigNumber> => {
   return bigNumberify(hc['round']);
 };
 const getTimeSecs = async (now_bn: BigNumber): Promise<BigNumber> => {
+  // TODO - this is not the right place to do invalidation, but I'm not sure where is...
+  viewCacheInvalidate(now_bn);
```

Comment:
Yea, this is not a good place to invalidate... we know that a block lasts ~5 seconds (except on our devnet) so we could just check the time of the cache and invalidate if it is more than 5 seconds in the past

Reply:
Is there a way to check whether we are on devnet?  You mentioned caching that expires with elapsed time OR when a new block is ready, but we could also just use a smaller timeout for devnet.

Reply:
We know we're on the devnet from `isIsolatedNetwork`